lib/ASN1/Universal/Integer.php

```diff
@@ -60,14 +65,13 @@ protected function getEncodedValue()
         $contentLength = $this->getContentLength();
 
         if ($numericValue < 0) {
```

Comment:
Hmm, I'll need to replace that with gmp_sign I suspect, but it should work then. I'll add some more tests for positive / negative numbers from 2^1 - 2^256

Main stuff to check is the twos complement stuff.. The first bit/byte is used a mask to indicate the sign. so AFAIK, given an n-bit integer, we can only save −(2^n−1) to +(2^(n−1) − 1).

Will send another commit for this today